azkaban-exec-server/src/main/java/azkaban/execapp/AzkabanExecServerModule.java

```diff
@@ -30,11 +37,49 @@
  */
 public class AzkabanExecServerModule extends AbstractModule {
 
+  private static final Logger logger = Logger.getLogger(AzkabanExecServerModule.class);
+  private static final String EVENT_REPORTER_CLASS_PARAM = "event.reporter.class";
+  private static final String EVENT_REPORTER_ENABLED = "event.reporter.enabled";
+
   @Override
   protected void configure() {
     bind(ExecutorLoader.class).to(JdbcExecutorLoader.class);
     bind(AzkabanExecutorServer.class).in(Scopes.SINGLETON);
     bind(TriggerManager.class).in(Scopes.SINGLETON);
     bind(FlowRunnerManager.class).in(Scopes.SINGLETON);
   }
+
+  @Inject
+  @Provides
+  public AzkabanEventReporter createAzkabanEventReporter(final Props props) {
+    final boolean eventReporterEnabled = props.getBoolean(EVENT_REPORTER_ENABLED, false);
+
+    if (!eventReporterEnabled) {
+      logger.info("Event reporter is not enabled");
+      return null;
+    }
+
+    final Class<?> eventReporterClass =
+        props.getClass(EVENT_REPORTER_CLASS_PARAM, null);
```

Comment:
How about providing a default "AzkabanKafkaAvroEventReporter" here so that you don't have to have a special case?

Reply:
Yes, good point. I will change that.

Reply:
done.